internal/servers/controller/testing.go

```diff
@@ -629,3 +638,71 @@ func (tc *TestController) AddClusterControllerMember(t *testing.T, opts *TestCon
 	}
 	return NewTestController(t, nextOpts)
 }
+
+// WaitForNextWorkerStatusUpdate waits for the next status check from a worker to
+// come in. If it does not come in within the default status grace
+// period, this function returns an error.
+func (tc *TestController) WaitForNextWorkerStatusUpdate(workerId string) error {
```

Comment:
Why are these two functions separate? One just calls the other -- why not just have a single function?

I looked and don't see it used anywhere else -- even tests only use the exported version.

Reply:
This was to avoid some issues with short-circuiting on context timeouts in the `select` at the top of the inner function's `for` loop. I've fixed this and by just wrapping the select in a function so we can `break` properly and the fix will be in the next PR.